metis-core/metis-core-service/src/test/java/eu/europeana/metis/core/service/RedirectionTest.java

```diff
@@ -0,0 +1,216 @@
+
+        final IndexToPublishPluginMetadata indexToPublishPluginMetadata = getIndexToPublishPluginMetadata(indexToPreviewPlugin);
+        indexToPublishPluginMetadata.setPerformRedirects(true);
+        final IndexToPublishPlugin indexToPublishPlugin = getIndexToPublishPlugin(indexToPublishPluginMetadata, getExecutionProgress());
+
+        final PluginWithExecutionId<ExecutablePlugin> indexToPublishPluginPluginWithExecutionId =
+                new PluginWithExecutionId<>("executionId", indexToPublishPlugin);
+        when(workflowExecutionDao.getLatestSuccessfulExecutablePlugin(anyString(), any(), anyBoolean()))
+                .thenReturn(indexToPublishPluginPluginWithExecutionId);
+
+        final PluginWithExecutionId<ExecutablePlugin> indexToPreviewPluginPluginWithExecutionId =
+                new PluginWithExecutionId<>("executionId", indexToPreviewPlugin);
+        when(dataEvolutionUtils.getRootAncestor(any()))
+                .thenReturn(indexToPreviewPluginPluginWithExecutionId)
+                .thenReturn(indexToPublishPluginPluginWithExecutionId);
+
+        final String datasetId = "datasetId";
+        final ObjectId objectId = new ObjectId();
+        final Workflow workflow = getWorkflow(datasetId, objectId, getIndexToPublishPluginMetadata(indexToPreviewPlugin));
+        final Dataset dataset = getTestDataset(datasetId);
+        final PluginWithExecutionId<ExecutablePlugin> predecessor = new PluginWithExecutionId<>("executionId", indexToPreviewPlugin);
+
+        final WorkflowExecution workflowExecution = workflowExecutionFactory.createWorkflowExecution(workflow, dataset, predecessor, priority);
+
+        final AbstractMetisPlugin<IndexToPublishPluginMetadata> abstractMetisPlugin = workflowExecution.getMetisPlugins().stream().findFirst().get();
+        assertTrue(abstractMetisPlugin.getPluginMetadata().isPerformRedirects());
+    }
+
+    @Test
+    void redirectionReviewWithPerformRedirectsWhenRedirectIdsPresent() throws BadContentException {
+        final int priority = 0;
+        final IndexToPreviewPluginMetadata indexToPreviewPluginMetadata = getIndexToPreviewPluginMetadata();
+        final IndexToPreviewPlugin indexToPreviewPlugin = getIndexToPreviewPlugin(indexToPreviewPluginMetadata, getExecutionProgress());
+
+        final IndexToPublishPluginMetadata indexToPublishPluginMetadata = getIndexToPublishPluginMetadata(indexToPreviewPlugin);
+        indexToPublishPluginMetadata.setPerformRedirects(true);
+        final IndexToPublishPlugin indexToPublishPlugin = getIndexToPublishPlugin(indexToPublishPluginMetadata, getExecutionProgress());
+
+        final PluginWithExecutionId<ExecutablePlugin> indexToPublishPluginPluginWithExecutionId =
+                new PluginWithExecutionId<>("executionId", indexToPublishPlugin);
+        when(workflowExecutionDao.getLatestSuccessfulExecutablePlugin(anyString(), any(), anyBoolean()))
+                .thenReturn(indexToPublishPluginPluginWithExecutionId);
+
```

Comment:
correct. the mock was there before and because of this condition(see below) mockito throws an exception of `Unnecessary stubbings detected.` meaning that `OR` in the first part of the expression is `true` (datasetUpdateSinceLatestPlugin) and it doesn't bothers to test the rest of the expression because of `OR` logic. 

```
 // In either of these situations, we perform a redirect.
performRedirect = datasetUpdatedSinceLatestPlugin || rootDiffersForLatestPlugin.getAsBoolean();
```
added the lenient(). prefix to the mock so is explicitly there as documentation.